packages/infra/lib/ihe-stack/ihe-gateway.ts

```diff
@@ -76,26 +74,47 @@ export function createIHEGateway(stack: Construct, props: IHEGatewayProps): void
     db,
     name: `${name}Inbound`,
     dnsSubdomain: "inbound",
-    httpPorts: [config.inboundPorts.patientDiscovery, config.inboundPorts.documentQuery],
+    pdPort: config.inboundPorts.patientDiscovery,
+    dqPort: config.inboundPorts.documentQuery,
+    drPort: config.inboundPorts.documentRetrieval ?? config.inboundPorts.documentQuery,
   });
 
-  const patientDiscoveryPort = config.inboundPorts.patientDiscovery;
-  const documentQueryPort = config.inboundPorts.documentQuery;
-  const documentRetrievalPort = config.inboundPorts.documentRetrieval ?? documentQueryPort;
-  const buildInboundAddress = (port: number) => `http://${iheGWAddressInbound}:${port}`;
-
-  addProxyRoute("/v1/patient-discovery", buildInboundAddress(patientDiscoveryPort), apiGateway);
-  addProxyRoute("/v1/document-query", buildInboundAddress(documentQueryPort), apiGateway);
-  addProxyRoute("/v1/document-retrieval", buildInboundAddress(documentRetrievalPort), apiGateway);
-}
+  // setup a private link so the API GW can talk to the ALB
+  const vpcLink = new apigwv2.VpcLink(stack, "IHEAPIGWVPCLink", {
+    vpc: props.vpc,
+  });
 
-function addProxyRoute(path: string, serverAddress: string, apiGateway: apigwv2.HttpApi) {
-  const integration = new HttpUrlIntegration(`${path}Integration`, serverAddress, {
-    method: apigwv2.HttpMethod.ANY,
+  apiGateway.addRoutes({
+    path: "/v1/patient-discovery",
+    integration: new HttpAlbIntegration(`IHEGWPDIntegration`, pdListener, {
+      vpcLink,
+      parameterMapping: new apigwv2.ParameterMapping().overwritePath(
+        apigwv2.MappingValue.custom(
+          "/Gateway/PatientDiscovery/1_0/NhinService/NhinPatientDiscovery?wsdl"
```

Comment:
I'm cool with the hardcoded weird urls, but just for viz, did we talk about trying to either change these or hide them behind configs?

Reply:
I can raise an issue for this, but prob not worth it since they're hidden internally, and will be gone after we migrate to OSS mirth anyways